services-camel/src/test/java/edu/unc/lib/cdr/fulltext/FulltextRouterTest.java

```diff
@@ -0,0 +1,189 @@
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_EVENT_TYPE;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.camel.BeanInject;
+import org.apache.camel.EndpointInject;
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.PropertyInject;
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
+import org.apache.camel.model.ModelCamelContext;
+import org.apache.camel.test.spring.CamelSpringTestSupport;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+import edu.unc.lib.dl.fcrepo4.PIDs;
+import edu.unc.lib.dl.fcrepo4.Repository;
+
+public class FulltextRouterTest extends CamelSpringTestSupport {
+	private static final String EVENT_NS = "http://fedora.info/definitions/v4/event#";
+	private static final String EVENT_TYPE = "org.fcrepo.jms.eventType";
+	private static final String IDENTIFIER = "org.fcrepo.jms.identifier";
+	private static final String RESOURCE_TYPE = "org.fcrepo.jms.resourceType";
+	private static final long TIMESTAMP = 1428360320168L;
+	private static final String USER_ID = "bypassAdmin";
+	private static final String USER_AGENT = "curl/7.37.1";
+	private static final String FILE_ID = "/file1";
+	private static final String EVENT_TYPES = EVENT_NS + "ResourceCreation";
+	private static final String ENHANCEMENT_ROUTE = "CdrServiceFulltextExtraction";
+	private static final String EXTRACTION_ROUTE = "HasFulltext";
+	
+	@PropertyInject(value = "fcrepo.baseUri")
+	private static String baseUri;
+	
+	@EndpointInject(uri = "mock:fcrepo")
+	protected MockEndpoint resultEndpoint;
+	
+	@BeanInject(value = "repository")
+	private Repository repo;
+	
+	@Produce(uri = "direct:start")
+	protected ProducerTemplate template;
+	
+	@Before
+	public void init() {
+		PIDs.setRepository(repo);
+		when(repo.getBaseUri()).thenReturn(baseUri);
+	}
+	
+	@Override
+	protected AbstractApplicationContext createApplicationContext() {
+		return new ClassPathXmlApplicationContext("/service-context.xml", "/fulltext-context.xml");
+	}
+	
+	@Test
+	public void testRouteStart() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(1);
+		
+		createContext(ENHANCEMENT_ROUTE);
+		
+		template.sendBodyAndHeaders("", createEvent(FILE_ID, EVENT_TYPES));
+
+		assertMockEndpointsSatisfied();
+	}
+	
+	@Test
+	public void testEventTypeFilter() throws Exception {
+		getMockEndpoint("mock:direct:fulltext.filter").expectedMessageCount(0);
+		
+		createContext(ENHANCEMENT_ROUTE);
+		
+		Map<String, Object> headers = createEvent(FILE_ID, EVENT_TYPES);
+		headers.put(EVENT_TYPE, "ResourceDeletion");
```

Comment:
If we are never changing the value passed to createEvent (EVENT_TYPES) and we aren't testing it in the class, you can probably remove it as a parameter to the createEvent method to avoid confusion like this.  I'm kind of wondering if we're supposed to be checking the other headers?  what do they exist for?